examples/vae_renyi.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+"""Variational auto-encoder for MNIST data.
+
+References
+----------
+http://edwardlib.org/tutorials/decoder
+http://edwardlib.org/tutorials/inference-networks
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import edward as ed
+from edward.inferences.renyi_divergence import Renyi_divergence
+import numpy as np
+import os
+import tensorflow as tf
+
+from edward.models import Bernoulli, Normal
+from edward.util import Progbar
+from keras.layers import Dense
+from scipy.misc import imsave
+from tensorflow.examples.tutorials.mnist import input_data
+
+DATA_DIR = "data/mnist"
+IMG_DIR = "img"
+
+if not os.path.exists(DATA_DIR):
+    os.makedirs(DATA_DIR)
+if not os.path.exists(IMG_DIR):
+    os.makedirs(IMG_DIR)
+
+ed.set_seed(42)
+
+M = 100  # batch size during training
+d = 2  # latent dimension
+alpha = 0.5     # alpha values for reny divergence
+n_samples = 5   # number of samples used to estimate the Renyi ELBO
+backward_pass = 'max'   # Back propagation style ('min', 'max' or 'full')
+
+# DATA. MNIST batches are fed at training time.
+mnist = input_data.read_data_sets(DATA_DIR)
+
+# MODEL
+# Define a subgraph of the full model, corresponding to a minibatch of
+# size M.
+z = Normal(loc=tf.zeros([M, d]), scale=tf.ones([M, d]))
+hidden = Dense(256, activation='relu')(z.value())
+x = Bernoulli(logits=Dense(28 * 28)(hidden))
+
+# INFERENCE
+# Define a subgraph of the variational model, corresponding to a
+# minibatch of size M.
+x_ph = tf.placeholder(tf.int32, [M, 28 * 28])
+hidden = Dense(256, activation='relu')(tf.cast(x_ph, tf.float32))
+qz = Normal(loc=Dense(d)(hidden),
+            scale=Dense(d, activation='softplus')(hidden))
+
+# Bind p(x, z) and q(z | x) to the same TensorFlow placeholder for x.
+inference = Renyi_divergence({z: qz}, data={x: x_ph})
```

Comment:
This code looks exactly the same as an older version of `vae.py` but only differs in this line. To keep the VAE versions better synced, could you add a comment suggesting that this is also an alternative in the existing `vae.py`?

Ideally, we'd like a specific application where `ed.RenyiDivergence` produces better results by some metric than alternatives. IIRC, the paper had some interesting results for a Bayesian neural net on some specific UCI data sets. That would be great to have and reproduce some of their results.

If you don't have time for this, we can leave it off for now and raise it as a Github issue post-merging this PR.